### DISCUSSION TITLE
Add a cancel event and/or command to the MAUI SearchBar control

### DISCUSSION THREAD

**Discussion by vsfeedback** [dotnet/maui#20995] [Ideas]
_This issue has been moved from [a ticket on Developer Community](https://developercommunity.visualstudio.com/t/Add-a-cancel-event-andor-command-to-the/10491917)._

---
The .NET MAUI SearchBar control (as of the release of Microsoft.Maui.Controls 8.0.0-rc2.9373) has no event or command associated with the cancel button, and neither the SearchCommand command nor the SearchButtonPressed event is invoked when the cancel button is tapped or the cancel (x) is tapped.  This makes it impossible to know when a user cancels the search, effectively making the control nearly useless unless you perform "live" searching as the user types text into the control.

---
#### Feedback Bot on 10/15/2023, 09:53 PM: 

(private comment, text removed)

**Comment by ideas2appsaustralia** [comment]
this is an excellent idea. I need to handle the "clear" searchbar text event when pressed. Maybe I can check when the searchbar text is empty by binding to TextChanged and trap the event that way or if this wont work create my own button that clear the searchbar text and then does other stuff i want. 

Yes, binding to "TextChanged" event of Searhbar captures the clear button press action. So I can then know when a user cancels the search.

Something like

 if (string.IsNullOrEmpty(SearchQuery))
	"do something";
In an ICommand that is bound to TextCHanged ofSearchbar worked
